Source/utils/stdcompat/shared_ptr_array.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+namespace devilution {
+
+// Apple Clang 12 has a buggy implementation that fails to compile `std::shared_ptr<T[]>(new T[size])`.
+#if __cplusplus >= 201611L && (!defined(__clang_major__) || __clang_major__ >= 13)
+template <typename T>
+using SharedPtrArray = std::shared_ptr<T[]>;

```

Comment:
This name is incredibly misleading to me. IMHO, it indicates "an array of shared pointers", but it is a "shared pointer to an array".

I suggest it be renamed for clarity:
```suggestion
using ArraySharedPtr = std::shared_ptr<T[]>;
```

Reply:
Done